components/script/dom/node.rs

```diff
@@ -757,22 +785,28 @@ impl Node {
         self.is_in_doc() && self.owner_doc().browsing_context().is_some()
     }
 
-    pub fn children(&self) -> NodeSiblingIterator {
-        NodeSiblingIterator {
-            current: self.GetFirstChild(),
+    pub fn children<T>(&self) -> SiblingIterator<T> where T: DerivedFrom<Node> {
+        let current = match self.GetFirstChild() {
+            Some(current) => Root::downcast::<T>(current),
+            None => None,
+        };
```

Comment:
The methods returning the iterators must be updated to use the new functions, e.g. this one should call `get_next_sibling(self)` to initialise the `current` field, this is probably why there are runtime errors with your current code.